File: java/arrays/905_SortArrayByParity.java
class Solution {
    public int[] sortArrayByParity(int[] A) {
        int L = A.length;
        int i = 0, j = L-1;
        int result[] = new int[L];
        for (int k=0; k<A.length; k++){
            if(A[k]%2==0){
                result[i] = A[k]; i++;
            }else{
                result[j] = A[k]; j--;
            }
        }
        return result;
    }
}

File: python/arrays/905_SortArrayByParity.py
class Solution:
    def sortArrayByParity(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        from collections import deque
        queue = deque();
        for i in A:
            if i%2==0: queue.appendleft(i);
            else:      queue.append(i);
        return list(queue);